src/transport.rs

```diff
@@ -68,7 +68,11 @@ impl WireNetwork {
         info!("Listening on: {}", socket.local_addr()?);
         loop {
             let mut bytes = [0; MAX_DATAGRAM_SIZE];
-            let (_, remote_address) = socket.recv_from(&mut bytes).await?;
+            let (_, remote_address) =
+                socket.recv_from(&mut bytes).await.map_err(|e| {
```

Comment:
This error does not seem to be handled. The function always returns an IP address